pop-api/integration-tests/src/fungibles/mod.rs

```diff
@@ -488,89 +445,74 @@ fn clear_metadata_works() {
-		// Asset is not live, i.e. frozen or being destroyed.
-		start_destroy_asset(addr.clone(), asset);
-		assert_eq!(
-			mint(addr.clone(), asset, BOB, amount),
-			Err(Module { index: 52, error: [16, 0] })
-		);
+		assert_eq!(mint(&addr, token, &BOB, Balance::MAX,), Err(Arithmetic(Overflow)));
+		// Token is not live, i.e. frozen or being destroyed.
+		pallet_assets_start_destroy(&addr, token);
+		assert_eq!(mint(&addr, token, &BOB, amount), Err(Module { index: 52, error: [16, 0] }));
 	});
 }
 
 #[test]
 fn burn_works() {
 	new_test_ext().execute_with(|| {
-		let _ = env_logger::try_init();
 		let addr = instantiate(CONTRACT, INIT_VALUE, vec![]);
 		let amount: Balance = 100 * UNIT;
 
-		// Asset does not exist.
-		assert_eq!(burn(addr.clone(), 1, BOB, amount), Err(Module { index: 52, error: [3, 0] }));
-		let asset = create_asset(ALICE, 1, 1);
-		// Bob has no tokens and thus pallet assets doesn't know the account.
-		assert_eq!(burn(addr.clone(), asset, BOB, 1), Err(Module { index: 52, error: [1, 0] }));
+		// Token does not exist.
+		assert_eq!(burn(&addr, 1, &BOB, amount), Err(Module { index: 52, error: [3, 0] }));
+		let token = pallet_assets_create(&ALICE, 1, 1);
+		// Bob has no tokens and therefore doesn't exist.
+		assert_eq!(burn(&addr, token, &BOB, 1), Err(Module { index: 52, error: [1, 0] }));
 		// Burning can only be done by the manager.
-		mint_asset(ALICE, asset, BOB, amount);
-		assert_eq!(burn(addr.clone(), asset, BOB, 1), Err(Module { index: 52, error: [2, 0] }));
-		let asset = create_asset_and_mint_to(addr.clone(), 2, BOB, amount);
-		// Asset is not live, i.e. frozen or being destroyed.
-		freeze_asset(addr.clone(), asset);
-		assert_eq!(
-			burn(addr.clone(), asset, BOB, amount),
-			Err(Module { index: 52, error: [16, 0] })
-		);
-		thaw_asset(addr.clone(), asset);
+		pallet_assets_mint(&ALICE, token, &BOB, amount);
+		assert_eq!(burn(&addr, token, &BOB, 1), Err(Module { index: 52, error: [2, 0] }));
+		let token = pallet_assets_create_and_mint_to(&addr, 2, &BOB, amount);
+		// Token is not live, i.e. frozen or being destroyed.
+		pallet_assets_freeze(&addr, token);
+		assert_eq!(burn(&addr, token, &BOB, amount), Err(Module { index: 52, error: [16, 0] }));
+		pallet_assets_thaw(&addr, token);
 		// Successful mint.
-		let balance_before_burn = Assets::balance(asset, &BOB);
-		assert_ok!(burn(addr.clone(), asset, BOB, amount));
-		let balance_after_burn = Assets::balance(asset, &BOB);
+		let balance_before_burn = Assets::balance(token, &BOB);
+		assert_ok!(burn(&addr, token, &BOB, amount));
+		let balance_after_burn = Assets::balance(token, &BOB);
 		assert_eq!(balance_after_burn, balance_before_burn - amount);
-		// Asset is not live, i.e. frozen or being destroyed.
-		start_destroy_asset(addr.clone(), asset);
-		assert_eq!(
-			burn(addr.clone(), asset, BOB, amount),
-			Err(Module { index: 52, error: [17, 0] })
-		);
+		// Token is not live, i.e. frozen or being destroyed.
+		pallet_assets_start_destroy(&addr, token);
```

Comment:
Instead of repeating the methods with `pallet_assets` as prefix like this, why don't we create a `pallet_assets` module and put those methods inside it?

Reply:
Much cleaner, thank you!